lib/src/Pos/Util/Servant.hs

```diff
@@ -133,15 +139,22 @@ class ModifiesApiRes t where
 -- 'ModifiesApiRes'.
 data VerbMod (modType :: *) api
 
-instance (HasServer (Verb mt st ct $ ApiModifiedRes mod a) ctx,
-          ModifiesApiRes mod) =>
+instance ( HasServer (Verb mt st ct $ ApiModifiedRes mod a) ctx
+         , ReflectMethod mt
+         , KnownNat st
+         , AllCTRender ct a
+         , ModifiesApiRes mod
+         ) =>
```

Comment:
This is a bit intimidating. Any chance we could document what's going on here and why we need those extra constraints ?

Reply:
@adinapoli-iohk 
This is because `hoistServerWithContext` is defined as the same as `hoistServerWithContext` for `Verb mt st ct a`, and we have `HasServer` constraint only for `Verb mt st ct (ApiModifierRes mod a)`. These constraints imply `HasServer (Verb mt st ct a)`.

Though I still have a couple of questions:
- Isn't it simpler and more clear to directly require `HasServer (Verb mt st ct a)` constraint (as you did everywhere else)?
- Is there actually any solid reason to not use `hoistServerWithContext` implementation for `Verb mt st ct (ApiModifier res a)`?

cc @Martoon-00

Reply:
> Isn't it simpler and more clear to directly require HasServer (Verb mt st ct a) constraint (as you did everywhere else)?

You're right, I'm switching to it..

> Is there actually any solid reason to not use hoistServerWithContext implementation for Verb mt st ct (ApiModifier res a)?
If I delegate to it, then I also need a function of signature `m a -> m (ApiModifiedRes mod a)`, and I have only roughly `Handler a -> Handler (ApiModifiedRes mod a)` which is important for errors catching.